### DISCUSSION TITLE
Getting Multiple Contact Force Vectors for a Single Capsule Geom/Body

### DISCUSSION THREAD

**Discussion by heleiduan** [google-deepmind/mujoco#113] [General]
My simulation model has a body with a single geom attached to it. However, in Mujoco Sim, when pressing 'C', I see two contacts showing up. Similarly with pressing 'F', I can see two force vectors coming from two end of the geom/body, potentially from the two-half spheres. I am able to get the single force vector applied onto the body. But how do I read the two distinct force vectors that are shown in the simulation? Thank you!

**Comment by yuvaltassa** [comment]
Please attach your model so we can inspect.

**Reply by heleiduan** [reply]
Here is the model file zipped. The force vectors are referring to the ones coming from the contact on the feet ("left-foot" at line 147 and "right-foot" at line 211) in the model. Thank you!
[model_file.zip](https://github.com/deepmind/mujoco/files/7907711/model_file.zip)

**Comment by yuvaltassa** [comment]
Your model has the unfortunate line `<material name='collision' rgba='0.1 0.5 0.9 0'/>`, so your collision geometries are hard-coded to be **invisible**. Set the alpha value to 1 and then add `group="1"` to this default: `<geom material='collision' contype='1' user='2'/>`. Now, clicking the '0' and '1' keys will respectively show/hide the visual and collision geometries.

**Reply by heleiduan** [reply]
Thank you! The collision geom for feet is a single capsule geom. We are able to get the single contact force vector using mj_contactForce() with the corresponding geom_id. But looking at the image when pressing 'F' and 'C', we can also see the two distinct force vectors/contacts at both ends of the capsule. How do we get the values for two distinct vectors? 
![contact](https://user-images.githubusercontent.com/4071750/150421625-ebd1d436-2500-4377-bb96-6d6fea9397af.png)

**Reply by yuvaltassa** [reply]
The id in [`mj_contactForce`](https://mujoco.readthedocs.io/en/latest/APIreference.html#mj-contactforce) is not a geom id but rather a contact id. You're supposed to iterate over the `data.contact` buffer and look at the id's of the the two geoms that are contacting.
